### ISSUE TITLE
[Export] output property type in XML

### ISSUE THREAD

**Issue by dkarlovi** [instride-ch/pimcore-data-definitions#125]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Branch?          | master

Specify property type (like `text`, `int` or similar) to allow not losing information when importing.

**Comment by dpfaffenbauer** [comment]
can you explain a bit more pls? the data-type comes from pimcore, so I don't quite get it.

**Comment by dkarlovi** [comment]
If you export and plan to import somewhere else, it's vital to keep the whole semantics

For example,
```xml
<property></property>
```

Is that meant to be `bool(false)`, `string(0) ""`, `int(0)` or `NULL`?

Having

```xml
<property type="string"></property>
```

removes that doubt.

**Comment by dpfaffenbauer** [comment]
true, but where to get the type? Pimcore Class Definition would be the best place right?

**Comment by dkarlovi** [comment]
I think so, yes.

**Comment by dkarlovi** [comment]
Also, maybe we can just export `gettype($value)`? Since the type will be determined by the interpreters too, not directly related to Pimcore's type.

**Comment by dpfaffenbauer** [comment]
yes, but some could be `null`, some could be `array` for the same property, since pimcore sometimes returns `null` for empty arrays. I think we should just use this: https://github.com/pimcore/pimcore/blob/master/models/DataObject/ClassDefinition/Data.php#L499

**Comment by dkarlovi** [comment]
Yes, but you might have transformed an array to a string by fetching only the first element? Also, some mappings will not have an "origin" (custom props).

The idea here is to keep the original data's type **as it was serialized** to XML. I think checking the final type should be good enough and cover almost all cases.

**Comment by dpfaffenbauer** [comment]
ok then, PR?

**Comment by dpfaffenbauer** [comment]
ping @dkarlovi

**Comment by dkarlovi** [comment]
Probably best to use a standards based solution like adding `xsi:nil` for null, etc.

We talked about adding an inline schema to enable the user to validate the file, this might be connected.

**Comment by dkarlovi** [comment]
http://www.herongyang.com/XML/XSD-Statements-Embedded-in-XML-File.html

**Comment by dkarlovi** [comment]
So, this would add a XML Schema instance namespace like this

```xml
<export xmlns:xsd="http://www.w3.org/2001/XMLSchema"
        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
    <property xsi:type="xsd:boolean">true</property>
</export>
```

https://www.w3.org/TR/xmlschema-1/#Instance_Document_Constructions
